Update meeds (#34)
exoplatform/developer.exoplatform.org: content/sources/workflow.adoc

eXo Platform product relies on a set of Git repositories :

> * https://github.com/Meeds-io[*meeds-io*]: Meeds-io association product Git organization. Meeds Lab is the main contributor of the source code provided by the association. eXo Platform is the main maintener of this source code (the eXo Maintenance Team is in charge to fix bug on maintenance versions)

Reviewer: IMO, here just need to describe the repository as the property of https://meed.io association with License LGPL v3 (Open Source and free to use). In addition we will need to remind that all members of the association benefits from this product and thus any feature/fix would affect the others. That's why we will need to communicate feature proposals and fixes through [MIPs](https://github.com/Meeds-io/MIPs) and [issues](https://github.com/Meeds-io/meeds/issues)

Reviewer 2: +1

Author: done

> * https://github.com/Meeds-io[*meeds-io*]: Meeds-io association product Git organization. Meeds Lab is the main contributor of the source code provided by the association. eXo Platform is the main maintener of this source code (the eXo Maintenance Team is in charge to fix bug on maintenance versions)

Reviewer 2: ```suggestion
* https://github.com/Meeds-io[*meeds-io*]: Meeds-io association product GitHub organization. Meeds Lab is the main contributor of the source code provided by the association. eXo Platform is the main maintainer of this source code (the eXo Maintenance Team is in charge of fixing bugs on maintenance versions)
```

> * `stable/*` : during Test Compaign of a minor or major release, when the FTs disn't started to work on new features yet.

Reviewer 2: ```suggestion
* `stable/*` : during Test Campaign of a minor or major release, when the FTs didn't start to work on new features yet.
```

> * `develop-meeds` and `develop-exo` : development branch for each association member

Reviewer: ```suggestion
* `develop-XXX` : development branch for each association member (by example, `develop-exo` for eXo Platform)
```

Author: done

> * `develop-XXX` : development branch for each association member (by example, `develop-exo` for eXo Platform)

Reviewer 2: ```suggestion
* `develop-XXX` : development branch for each association member (for example, `develop-exo` for eXo Platform)
```

== Branching model

Branching model are 6 kinds of branch :

* `develop` : Develop branch contains the latest delivered and stable development changes.

> * `develop-xxx` : the develop branch for each Meeds association member. This branch is the branch deployed in production by the CI/CD. For example, each night, Builders application is deployed from branch develop-meeds. eXo Tribe is deployed from branch develop-exo. Theses branch are only present on meeds-io repositories.

Reviewer 2: ```suggestion
* `develop-xxx` : the develop branch for each Meeds Association's member. This branch is the branch deployed in production by the CI/CD. For example, each night, Builders application is deployed from branch develop-meeds. eXo Tribe is deployed from branch develop-exo. Theses branch are only present on meeds-io repositories.
```

> * `feature/xxx` : Feature branches are dedicated branch for one big feature (lot of commits), "xxx" is the feature name. Theses branches can be created from develop-meeds or develop-exo for meeds repositories, and from develop for exoplatform repositories.

Reviewer 2: ```suggestion
* `feature/xxx` : Feature branches are dedicated to a single large feature (lots of commits), "xxx" is the feature name. These branches can be created from develop-meeds or develop-exo for meeds repositories, and from develop for exoplatform repositories.
```

=== Develop Branch

> For meeds-io repositories, develop branch is the branch containing all finished and tested features.

Reviewer 2: ```suggestion
For meeds-io repositories, develop branch is the branch containing all completed and properly tested features.
```

> For exoplatform repositories, develop branch is the production ready branch which is, in general case, the CI/CD production branch.

Reviewer 2: ```suggestion
For exoplatform repositories, develop branch is the production-ready branch which is, in general case, the CI/CD production branch.
```

It contains the latest delivered development changes. This is our backbone where all the different fix and new feature are mixed with each other.

> For meeds-io repositories, when members finish a feature, they merge it on develop branch from their develop-xxx branch. Then, on a regular basis, a rebase is done to propagate these new changes on other members' develop branch :

Reviewer 2: ```suggestion
For meeds-io repositories, when members complete a feature, they merge it on develop branch from their develop-xxx branch. Then, on a regular basis, a rebase is done to propagate these new changes on other members' develop branch :
```

> For meeds-io repositories, when a member finish a feature, he merges it on develop branch from his develop-xxx branch. Then, on a regularly base, a rebase is done to propagate theses new developments on other members develop branch :

Reviewer 2: ```suggestion
For meeds-io repositories, when members finish a feature, they merge it on develop branch from their develop-xxx branch. Then, on a regular basis, a rebase is done to propagate these new changes on other members' develop branch :
```

[[git-workflow-develop-branch]]
.Git Workflow - Rebase Develop Branch

> image::git-worflow-rebase-develop.png[Git Workflow - Rebase Develop Branch, 800, align=center, link={imagesdir}/git-worflow-rebase-develop.png]

Reviewer: We will have to review this workflow since another proposition was made and applied by @hbenali

=== Feature Branch

Feature branches are dedicated branch to develop a new feature.

> On meeds-io repositories, feature branches are created from develop-xxx branch (develop-exo for and eXo Platform feature).

Reviewer 2: ```suggestion
On meeds-io repositories, feature branches are created from develop-xxx branch (develop-exo for and eXo Platform features).
```

> On meeds-io repositories, feature branches are created from develop-xxx branch (develop-meeds if it is a Meeds Lab feature, or develop-exo if it is an eXo Platform feature).

Reviewer: no need to specify `Meeds Lab` processes here, just eXo's ones (I prefer not having the term `Meeds Lab` inside eXo Platform Dev guide in fact). The internal processes of Meed Lab can change anytime (this will depends on technical decisions inside Meeds Lab).

Author: done

> On exoplatform repositories, features branches are created from develop branch.

Reviewer 2: ```suggestion
On exoplatform repositories, feature branches are created from develop branch.
```

[[git-workflow-feature-branch]]
.Git Workflow - Meeds Feature Branch
image::git-workflow-meeds-feature-branch.png[Git Workflow - Meeds Feature Branch, 800, align=center, link={imagesdir}/git-workflow-meeds-feature-branch.png]

> In meeds repositories, when a User Story (US) is finished, it is backported to develop-xxx branch. So that, the nightly deployment will push this US in production for the association member which develop it, without impacting other association members.

Reviewer 2: ```suggestion
In meeds repositories, when a User Story (US) is completed, it is backported to develop-xxx branch. So that, the nightly deployment will attempt to push the feature in production for the association member which develops it, without impacting other association members.
```
